src/source/tile.js

```diff
@@ -527,6 +531,12 @@ class Tile {
             if (!sourceLayer || !sourceLayerStates || Object.keys(sourceLayerStates).length === 0) continue;
 
             bucket.update(sourceLayerStates, sourceLayer, availableImages, this.imageAtlas && this.imageAtlas.patternPositions || {});
+            if (bucket instanceof LineBucket || bucket instanceof FillBucket || bucket instanceof FillExtrusionBucket || bucket instanceof CircleBucket) {
```

Comment:
Could you move this as a function on a bucket, instead of all of these instanceof checks?

Reply:
also is FillExtrusion right here? because thats not draped?

Reply:
The instance checks are necessary for Flow. Really we're just checking that it's not a SymbolBucket, since SymbolBucket doesn't have the `programConfigurations`.

Good point about draping, should this just be `LineBucket` and `FillBucket`?

Reply:
well symbols are draped in certain situations 😅 (map aligned)?

Reply:
😓 Hmmm... If this bug exists for symbols then this fix won't work as is. Though we could bypass the check for `bucket.programConfigurations.needsUpload` and clear the cache for all tiles.

Do we have an examples of `feature-state` on a symbol layer?

Reply:
Checked that symbols are not yet draped in any situation on gl-js, pending https://github.com/mapbox/mapbox-gl-js/issues/10365. The draped layer list is https://github.com/mapbox/mapbox-gl-js/blob/main/src/style/style.js#L116.

So this check should be good now?

Reply:
Yes!
Tested that symbol layers don't have this bug and that the fix works with line layers as well as fill layers.